### PR TITLE
Refactor paths

### PR DESCRIPTION
**Pull request by bdfinst**
## Description
To prepare the repo for documentation migration and to improve the discoverability of scripts for generating docs, I've renamed `hack/` to `scripts/` and moved the `.templates/` folder with the JSFH configs into `scripts/` path.

This will make doc conversion much cleaner and make it easier to find things.

### CODE REVIEW

**Review by Racer159 — CHANGES_REQUESTED** [review]

File: scripts/create-zarf-schema.sh [hunk 1]
```diff
@@ -1,10 +1,10 @@
 #!/usr/bin/env sh
 
 # Create the json schema for the zarf.yaml
-go run main.go internal config-schema > zarf.schema.json
+go run main.go internal config-schema >zarf.schema.json
```
[the remark this answers is not in the document]
Comment (bdfinst) [reply scripts/create-zarf-schema.sh:4]: Must be a  setting in my IDE. I'll check.

File: scripts/syft.tmpl
[the remark this answers is not in the document]
Comment (bdfinst) [reply scripts/syft.tmpl:1]: Done. FYI, `syft.tmpl` is not referenced anywhere.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -3,22 +3,22 @@ repos:
     rev: v4.0.1
     hooks:
       - id: check-added-large-files
-        args: ["--maxkb=1024"]
```
[the remark this answers is not in the document]
Comment (bdfinst) [reply .pre-commit-config.yaml]: Due to VS Code setting. Might be a good idea to add a project config file to control that.

File: scripts/.templates/jsfh-config.json [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+  "custom_template_path": "scripts/.templates/jsonschemaforhumans/base.md",
+  "template_md_options": {
+    "badge_as_image": true,
+    "show_heading_numbers": false
+  },
+  "show_toc": false,
+  "show_breadcrumbs": false,
+  "footer_show_time": false,
+  "deprecated_from_description": true
+}
```
[the remark this answers is not in the document]
Comment (bdfinst) [reply scripts/.templates/jsfh-config.json]: Bug in VS Code. It's not respecting the .editorconfig setting. I had to explicitly set my IDE to add it. Terrible.

File: .pre-commit-config.yaml
Comment (Racer159) [inline .pre-commit-config.yaml:1]: this file doesn't need to change

File: CODEOWNERS
Comment (Racer159) [inline CODEOWNERS:1]: The codeowners file should not change

**Review by Racer159 — CHANGES_REQUESTED** [review]
You will need to run `make docs-and-schema` to update the docs to reference `hack`

**Review by Racer159 — APPROVED** [review]
lgtm

**Pull request merged**